src/ORM/DataObject.php

```diff
@@ -2725,6 +2725,13 @@ public function dbObject($fieldName)
             ? $this->record[$fieldName]
             : null;
 
+        if (!$value && isset($this->record[$fieldName . '_Lazy'])) {
+            $tableClass = $this->record[$fieldName . '_Lazy'];
+            $this->loadLazyFields($tableClass);
+
+            $value = isset($this->record[$fieldName]) ? $this->record[$fieldName] : null;
```

Comment:
`!$value` matches false. would suggest using  `!isset($this->record[$fieldName])` in this if block instead, and moving the value-setter on line 2724 to below this if clause. then line 2732 isn't needed.